src/routes/recovery/recovery.controller.spec.ts

```diff
@@ -0,0 +1,138 @@
+    alertsAccount = configurationService.get('alerts.account');
+    alertsProject = configurationService.get('alerts.project');
+    networkService = moduleFixture.get(NetworkService);
+
+    app = await new TestAppProvider().provide(moduleFixture);
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  describe('POST add recovery module for a Safe', () => {
+    it('Success', async () => {
+      const addRecoveryModuleDto = addRecoveryModuleDtoBuilder().build();
+      const chainId = faker.string.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+
+      networkService.post.mockImplementation((url) =>
+        url ===
+        `${alertsUrl}/api/v2/accounts/${alertsAccount}/projects/${alertsProject}/contracts`
+          ? Promise.resolve({ status: 200 })
+          : Promise.reject(`No matching rule for url: ${url}`),
+      );
+
+      await request(app.getHttpServer())
+        .post(`/v1/chains/${chainId}/safes/${safeAddress}/recovery`)
+        .send(addRecoveryModuleDto)
+        .expect(200);
+    });
+
+    it('should get a validation error', async () => {
+      const addRecoveryModuleDto = addRecoveryModuleDtoBuilder().build();
+      const chainId = faker.string.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+
+      await request(app.getHttpServer())
+        .post(`/v1/chains/${chainId}/safes/${safeAddress}/recovery`)
+        .send(omit(addRecoveryModuleDto, 'moduleAddress'))
+        .expect(500)
+        .expect({ message: 'Validation failed', code: 42, arguments: [] });
+    });
+
+    it('Should return the alerts provider error message', async () => {
+      const addRecoveryModuleDto = addRecoveryModuleDtoBuilder().build();
+      const chainId = faker.string.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+
+      networkService.post.mockImplementation((url) =>
+        url ===
+        `${alertsUrl}/api/v2/accounts/${alertsAccount}/projects/${alertsProject}/contracts`
+          ? Promise.reject({
+              data: { message: 'Malformed body', status: 400 },
+              status: 400,
+            })
+          : Promise.reject(`No matching rule for url: ${url}`),
+      );
+
+      await request(app.getHttpServer())
+        .post(`/v1/chains/${chainId}/safes/${safeAddress}/recovery`)
+        .send(addRecoveryModuleDto)
+        .expect(400)
+        .expect({
+          message: 'Malformed body',
+          code: 400,
+        });
+    });
+
+    it('Should fail with An error occurred', async () => {
+      const addRecoveryModuleDto = addRecoveryModuleDtoBuilder().build();
+      const chainId = faker.string.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+
+      networkService.post.mockImplementation((url) =>
+        url ===
+        `${alertsUrl}/api/v2/accounts/${alertsAccount}/projects/${alertsProject}/contracts`
+          ? Promise.reject({ status: 503 })
```

Comment:
What is the reason behind testing 503 and not other error codes?

(not saying that we should test each error code but I'm curious about the 503 one)

Reply:
I used the [`delegates.controller.spec` as a reference](https://github.com/safe-global/safe-client-gateway/blob/recovery-route/src/routes/delegates/delegates.controller.spec.ts#L239) for this test. Would you suggest a different error code?

Reply:
Any client or server error from the external source should work for this test so you can have something like:

```typescript
const statusCode = faker.internet.httpStatusCode({ types: ['clientError', 'serverError'] });
```

and then use this status code in the assertions.

Reply:
I've changed it in 1b7d94a3cade2cdf3d86e5c12f982c2099259eec. I will alter the other tests to use that in a separate PR too.